flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/dataformat/Timestamp.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.dataformat;
+
+import org.apache.flink.util.Preconditions;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+
+/**
+ * Immutable SQL TIMESTAMP type with nanosecond precision.
+ *
+ * <p>This class is composite of a millisecond and nanoOfMillisecond. The millisecond part
+ * holds the integral second and the milli-of-second. The nanoOfMillisecond holds the
+ * nano-of-millisecond, which should between 0 - 999_999.
+ */
+public class Timestamp implements Comparable<Timestamp> {
```

Comment:
Can we find another name, it's the same with `java.sql.Timestamp` and will confuse the reader a lot when scanning source codes.

Reply:
I think Timestmap is OK for a internal class, actually hive also introduces a Timestamp class.
[Hive Timestamp](https://github.com/apache/hive/blob/master/common/src/java/org/apache/hadoop/hive/common/type/Timestamp.java)

Reply:
I'm not saying this name violates some rules, but changing to another can help us reading the codes, if we have some other options.

Reply:
For example, if we want to expose a get method for timestamp with `getTimestamp`, i would say more than 90% would think this is a `java.sql.Timestamp`

Reply:
What about DateTime?

Reply:
`DateTime` is a good choice, Joda also expose `DateTime` as its core API. However, if we introduced a zoned class, `ZonedDateTime` and `OffsetDateTime` have already been used by JDK.

Reply:
Maybe we can use `ZonedTimestmap` if we have no better choice. It is just not so consistent between `ZonedTimestamp` and `DateTime`.

Reply:
Another choice: DataTimeWithZone

Reply:
We can use Hive-style `DateTimeTZ` as TIMESTAMP_WITH_TIME_ZONE